### ISSUE TITLE
Add ruff as an automatic linter

### ISSUE THREAD

**Issue by johnjasa** [NREL/H2Integrate#50]
# Add ruff as an automatic linter

We should add a pre-commit hook for linting GreenHEART to streamline and simplify code coming from disparate sources.

## Proposed solution

[Ruff](https://github.com/astral-sh/ruff) is fast, easy-to-setup, and well-accepted.
I'll defer to @RHammond2 about best settings though it should be pretty straightforward.

**Comment by bayc** [comment]
Agreed! We setup pre-commit hooks in the electrolyzer repo (inspired by FLORIS) that we could pull from as well: https://github.com/NREL/electrolyzer
